src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -19,50 +18,104 @@
   private final double RETRACT_SPEED = -1.0;
 
   private final SmartMotorController motor;
-  private final BooleanSupplier upperLimit;
-  private final BooleanSupplier lowerLimit;
 
-  private final Modifier upperModifier;
-  private final Modifier lowerModifier;
-  private final ModifierGroup modifiers;
+  private final Modifier validatorLimit;
+  private final ModifierGroup switchLimit;
+  private final ModifierGroup limit;
 
   public Climber(TelescopeMap map) {
+    Modifier upperLimit = Modifier.upperLimit(map.getUpperLimit());
+    Modifier lowerLimit = Modifier.lowerLimit(map.getLowerLimit());
     motor = map.getMotor();
-    upperLimit = map.getUpperLimit();
-    lowerLimit = map.getLowerLimit();
-    upperModifier = Modifier.upperLimit(upperLimit);
-    lowerModifier = Modifier.lowerLimit(lowerLimit);
-    modifiers = new ModifierGroup(upperModifier, lowerModifier);
+    switchLimit = new ModifierGroup(upperLimit, lowerLimit);
+    validatorLimit = Modifier.unless(motor::errored);
+    limit = new ModifierGroup(upperLimit, lowerLimit, validatorLimit);
   }
 
-  public CommandBase move(DoubleSupplier speed) { // Move motor with variable speed that is affected by limit switches
-    return cmd("Move").onInitialize(() -> {
-      motor.set(modifiers.run(speed.getAsDouble()));
-    }).onExecute(() -> {
+  // Move the motor based off a variable speed
+  public CommandBase move(DoubleSupplier speed) {
+    return cmd("Move").onExecute(() -> {
+      motor.set(limit.run(speed.getAsDouble()));
+      SmartDashboard.putNumber("Climber Speed", speed.getAsDouble());
+    }).onEnd((interrupted) -> {
+      motor.set(0.0);
+    });
+  }
+
+  // Move the motor based on a variable speed and stop when validator fails
+  public CommandBase moveCurrent(DoubleSupplier speed) {
+    return cmd("Move With Current Limit").onExecute(() -> {
+      motor.set(validatorLimit.applyAsDouble(speed.getAsDouble()));
+      SmartDashboard.putNumber("Climber Speed", speed.getAsDouble());
+    }).finishedWhen(motor::errored).onEnd((interrupted) -> {
+      motor.set(0.0);
+    });
+  }
+
+  // Move the motor with variable speed and stop when limit switches are hit
+  public CommandBase moveLimit(DoubleSupplier speed) {
+    return cmd("Move With Limits").onExecute(() -> {
+      motor.set(switchLimit.run(speed.getAsDouble()));
       SmartDashboard.putNumber("Climber Speed", speed.getAsDouble());
     }).onEnd((interrupted) -> {
       motor.set(0.0);
     });
   }
 
   public CommandBase extend() {
-    return startEnd("Extend", () -> {
-      motor.set(upperModifier.applyAsDouble(EXTEND_SPEED));
-    }, () -> {
+    return cmd("Extend").onExecute(() -> {
```

Comment:
Can these three and their equivalent `retract` commands be abstracted at all? They look very similar, except for the speed.